### PR TITLE
fix: RepackStep must use the same KMS key as the Model

### PR DESCRIPTION
**Pull request by Guillem96**
*Description of changes:*

All repack steps forward the `output_kms_key` of the model that they are repacking.

*Testing done:*

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/workflow/step_collections.py [hunk 1]
```diff
@@ -400,6 +400,7 @@ def __init__(
                 security_group_ids=estimator.security_group_ids,
                 description=description,
                 display_name=display_name,
+                output_kms_key=estimator.output_kms_key,
```
Comment (qidewenwhen) [inline src/sagemaker/workflow/step_collections.py:403]: Thanks so much for helping to fix the issue!

By adding this line, you're updating the `EstimatorTransformer` behavior. 
Can we make similar changes to the TWO `_RepackModelStep` objects created in `RegisterModel` as well? Thanks!

Comment (Guillem96) [reply src/sagemaker/workflow/step_collections.py:403]: Hi @qidewenwhen, I think you are already providing the output_kms_key in the Register Step as a kwarg. It was only kssing on the Transformer and in the ModelStep.

Can you please check it again?

**Review by qidewenwhen — APPROVED** [review]

File: src/sagemaker/workflow/step_collections.py (hunk 1 above)
Comment (qidewenwhen) [reply src/sagemaker/workflow/step_collections.py:403]: Yea you're right. It's already covered here https://github.com/aws/sagemaker-python-sdk/blob/849ed29a7f1ff9f1526b23902ad7db7c384126a9/src/sagemaker/workflow/step_collections.py#L171 in the `RegisterModel`.
Again appreciate your help!

Will reach out to @mufaddal-rohawala to review and merge the PR in Monday.

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run all

**Review by navinsoni — COMMENTED** [review]
/bot run all

**Comment by qidewenwhen** [comment]
The flakey slow test failed due to unrelated issues. Re-ran the slow test

**Comment by qidewenwhen** [comment]
After retry, the slow test failed due to a different issue - `test_default_right_size_and_deploy_unregistered_base_model`.
@Guillem96 could you please rebase your change on top of the latest master branch?
The test issue seems to be fixed in https://github.com/aws/sagemaker-python-sdk/commit/9e5dca6a43de32fe4274f1ea07d1a65ee4ece795

**Review by navinsoni — COMMENTED** [review]
/bot run all

**Pull request merged**